src/synthesizer/parametric/stars.py

```diff
@@ -502,6 +511,10 @@ def generate_lnu(
             axis=0,
         )
 
+        # Apply the wavelength mask if provided
+        if lam_mask is not None:
+            spectra[~lam_mask] = 0.0
```

Comment:
Could we not be a bit cleverer here and limit the grid spectra with the mask in `prepare_sed_args` that is fed to the extension. That way we never do the calculation on the masked elements, instead we just add them into a zeros array here to ensure the shape is consistent (like when we apply a particle mask).

Reply:
See comments on all other instances :) feel free to implement